File: Extras/ordenação_insercao.py
from random import randint
lista =  list()

def createList(tamanho):
    for x in range(1, tamanho):
        randNumber = randint(0, 50)
        if randNumber not in lista:
            lista.append(randNumber)
        x
    print(lista)
    return lista


def insertionSort(newList):
    n = len(newList)

    for j in range(n):
        chave = newList[j]

        i = j - 1

        while i >= 0 and newList[i] > chave:
            newList[i + 1] = newList[i]
            i = i - 1
        newList[i + 1] = chave

    return newList

print(insertionSort(createList(20)))

File: Extras/ordenação_insercao.dart
import 'dart:math';

List lista = [];

randonNumbers(int tamanho) {
  var rng = Random();
  for (var i = 0; i < tamanho; i++) {
    lista.add(rng.nextInt(100));
  }
  return lista;
}

void main(List<String> args) {
  print(randonNumbers(10));
  print(insertionSort(lista));
}

insertionSort(List newList) {
  int n = newList.length;
  for (var j = 0; j < n; j++) {
    var chave = newList[j];
    var i = j - 1;

    while (i >= 0 && newList[i] > chave) {
      newList[i + 1] = newList[i];
      i = i - 1;
    }
    newList[i + 1] = chave;
  }
  return newList;
}
